src/main/java/pl/allegro/tech/opel/OpelParser.java

```diff
@@ -5,10 +5,11 @@
 import org.parboiled.annotations.BuildParseTree;
 import org.parboiled.annotations.SuppressSubnodes;
 
+import java.lang.invoke.MethodHandles;
 import java.math.BigDecimal;
 
 @BuildParseTree
-class OpelParser extends BaseParser<OpelNode> {
+public class OpelParser extends BaseParser<OpelNode> {
```

Comment:
why it has to be public?

Reply:
this is a long story.... :P
jdk16 changed warning of illegal module access reflection to runtime error.
Parboiled is based on this behaviour :( I've made this branch to test my fix. Fortunately parboiled author came along and he is willing to merge my fix ;) package classes cannot be accessed in the way parboild works (org.parboiled cannot access pl.allegro.tech.opel class)

Reply:
wow!